src/app/pages/swap/components/swap-selected-asset-from.tsx

```diff
@@ -0,0 +1,95 @@
+import { FiChevronDown } from 'react-icons/fi';
+
+import { Box, Stack, Text } from '@stacks/ui';
+import { useField, useFormikContext } from 'formik';
+
+import { isUndefined } from '@shared/utils';
+
+import { formatMoneyWithoutSymbol } from '@app/common/money/format-money';
+import { SelectedAssetField } from '@app/components/forms/selected-asset-field';
+import { Flag } from '@app/components/layout/flag';
+import { SpaceBetween } from '@app/components/layout/space-between';
+
+import { useAmountAsFiat } from '../hooks/use-amount-as-fiat';
+import { SwapFormValues } from '../hooks/use-swap';
+import { tempExchangeRate } from '../hooks/use-swappable-assets';
+import { SwapAmountField } from './swap-amount-field';
+import { SwapSelectedAssetLayout } from './swap-selected-asset.layout';
+
+const sendingMaxCaption = 'Using max available';
+const sendingMaxTooltip = 'When sending max, this amount is affected by the fee you choose.';
+
+const maxAvailableCaption = 'Max available in your balance';
+const maxAvailableTooltip =
+  'Amount of funds that is immediately available for use, after taking into account any pending transactions or holds placed on your account by the protocol.';
+
+const sendAnyValue = 'Send any value';
+
+interface SwapSelectedAssetFromProps {
+  isSendingMax: boolean;
+  onChooseAsset(): void;
+  onSetIsSendingMax(value: boolean): void;
+  title: string;
+}
+export function SwapSelectedAssetFrom({
+  isSendingMax,
+  onChooseAsset,
+  onSetIsSendingMax,
+  title,
+}: SwapSelectedAssetFromProps) {
+  const { setFieldValue, validateForm, values } = useFormikContext<SwapFormValues>();
+  const [amountField, _, amountFieldHelpers] = useField('swapAmountFrom');
+  const [assetField] = useField('swapAssetFrom');
+
+  const amountAsFiat = useAmountAsFiat(assetField.value.balance, amountField.value);
+
+  const formattedBalance = formatMoneyWithoutSymbol(assetField.value.balance);
+
+  async function onSetMaxBalanceAsAmountToSwap() {
+    if (isUndefined(values.swapAssetTo)) return;
+    onSetIsSendingMax(!isSendingMax);
+    const value = isSendingMax ? '' : formattedBalance;
+    amountFieldHelpers.setValue(value);
+    // TODO: Need exchange rate here from asset
+    await setFieldValue('swapAmountTo', Number(value) * tempExchangeRate);
+    await validateForm();
+  }
+
+  return (
```

Comment:
Feel like more layout can be abstract to this component. Why not use slots? Where you pass JSX in as props to put a certain piece in part of the HTML tree?

e.g.

```ts
    <SwapSelectedAssetLayout
      caption={isSendingMax ? sendingMaxCaption : maxAvailableCaption}
      onClickHandler={onSetMaxBalanceAsAmountToSwap}
      title={title}
      tooltipLabel={isSendingMax ? sendingMaxTooltip : maxAvailableTooltip}
      value={isSendingMax ? sendAnyValue : formattedBalance}
      swapAmountInput={
        <SwapAmountField
          amountAsFiat={amountAsFiat}
          isDisabled={isUndefined(values.swapAssetTo)}
          name="swapAmountFrom"
        />
      }
    >
```